tests/asgi/test_query.py

```diff
@@ -12,6 +12,13 @@ def test_simple_query(schema, test_client):
     assert response.json() == {"data": {"hello": "Hello world"}}
 
 
+def test_fails_when_request_body_has_invalid_json(test_client):
+    response = test_client.post(
+        "/", data='{"qeury": "{__typena"', headers={"content-type": "application/json"}
+    )
+    assert response.status_code == 400
```

Comment:
Not specific to this PR, but should we have a dedicated collection of tests for HTTP response codes? Is it worth it -- how many different situations are there?

Reply:
@BryceBeagle good point. @jkimbo mentioned in #1214 the GraphQL over HTTP spec. We should probably review our http views and make sure they comply with the spec and have sufficient tests, including tests response codes.